src/miner_util.erl

```diff
@@ -104,16 +104,30 @@ metadata_fun() ->
         Map = blockchain_worker:signed_metadata_fun(),
         case application:get_env(miner, mode, gateway) of
             validator ->
+                RPC =
+                    case lists:filter(fun libp2p_transport_tcp:is_public/1,
```

Comment:
maybe better to use libp2p_transport:sort_addrs_with_keys/2 which will put any public ip at the head ?